spec/lib/rake/extensiontask_spec.rb

```diff
@@ -620,6 +620,98 @@
+                .and_return(rbconf)
+            )
+          end
+        end
+
+        allow(Gem).to receive_message_chain(:configuration, :verbose=).and_return(true)
+
+        spec = Gem::Specification.new do |s|
+          s.name = 'my_gem'
+          s.platform = Gem::Platform::RUBY
+          s.extensions = ['ext/somegem/extconf.rb']
+        end
+
+        cross_specs = {} # platform => spec
+        Rake::ExtensionTask.new("extension_one", spec) do |ext|
+          ext.cross_platform = platforms
+          ext.cross_compile = true
+          ext.cross_compiling do |cross_spec|
+            cross_specs[cross_spec.platform.to_s] = cross_spec
+          end
+        end
+        platforms.each do |platform|
+          Rake::Task["native:my_gem:#{platform}"].execute
+        end
+
+        cross_specs["arm64-darwin"].required_rubygems_version.should eq(Gem::Requirement.default)
+        cross_specs["arm-linux"].required_rubygems_version.should eq(Gem::Requirement.default)
+
+        if Gem::Version.new(Gem::VERSION) >= Gem::Version.new("3.3.22")
+          expected_rubygems_version = Gem::Requirement.new([">= 3.3.22"])
+          cross_specs["arm-linux-gnu"].required_rubygems_version.should eq(expected_rubygems_version)
+          cross_specs["arm-linux-musl"].required_rubygems_version.should eq(expected_rubygems_version)
+        end
+      end
+
+      it "should append to required_rubygems_version if it's set" do
+        platforms = ["arm-linux-musl"]
+        ruby_cc_versions = ["3.3.0"]
+        ENV["RUBY_CC_VERSION"] = ruby_cc_versions.join(":")
+
+        ruby_cc_versions.each do |ruby_cc_version|
+          platforms.each do |platform|
+            rbconf = "/rubies/#{ruby_cc_version}/rbconfig.rb"
+            allow_any_instance_of(Rake::CompilerConfig).to(
+              receive(:find)
+                .with(ruby_cc_version, platform)
+                .and_return(rbconf)
+            )
+          end
+        end
+
+        allow(Gem).to receive_message_chain(:configuration, :verbose=).and_return(true)
+
+        spec = Gem::Specification.new do |s|
+          s.name = 'my_gem'
+          s.platform = Gem::Platform::RUBY
+          s.extensions = ['ext/somegem/extconf.rb']
+          s.required_rubygems_version = "!= 3.4.1" # keep this around
+        end
+
+        actual_cross_spec = nil
+        Rake::ExtensionTask.new("extension_one", spec) do |ext|
+          ext.cross_platform = platforms
+          ext.cross_compile = true
+          ext.cross_compiling do |cross_spec|
+            actual_cross_spec = cross_spec
+          end
+        end
+        platforms.each do |platform|
+          Rake::Task["native:my_gem:#{platform}"].execute
+        end
+
+        expected_rubygems_version = Gem::Requirement.new(["!= 3.4.1", ">= 3.3.22"])
+
+        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
+          actual_cross_spec.required_rubygems_version.should eq(Gem::Requirement.new(["!= 3.4.1"]))
+        else
```

Comment:
Could you reduce(?) scope?

```suggestion
        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
          actual_cross_spec.required_rubygems_version.should eq(Gem::Requirement.new(["!= 3.4.1"]))
        else
          expected_rubygems_version = Gem::Requirement.new(["!= 3.4.1", ">= 3.3.22"])
```

Reply:
I'm not sure what you mean. I think you mean make the test smaller or simpler?

I've tried to simplify this test slightly by removing the array of platforms. Is that helpful? Can you explain more what you mean?

Reply:
Ah, if you mean "move `expected_rubygems_version` after the `else` then I have done that.

Reply:
Done.